addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1202,7 +1203,20 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
      */
     _getDependencyEl(fieldEl = this.$target[0]) {
         const dependencyName = fieldEl.dataset.visibilityDependency;
-        return this.formEl.querySelector(`.s_website_form_input[name="${dependencyName}"]`);
+        return this.formEl.querySelector(`.s_website_form_input[name="${this._getNameForSelector(dependencyName)}"]`);
+    },
+    /**
+     * Replace all the occurrence of `"` by `\\"` and `'` by `\\'`. This is
+     * needed in order to be able to do perform a querySelector on the return
+     * string.
+     * @param {String} name
+     * @returns {String}
+     */
+    _getNameForSelector(name) {
+        if (name) {
```

Comment:
It is to avoid `undefined` value :+1: For example: -Drop a "Form" snippet on the website.
-Click on a field.
=> The argument of `_getNameForSelector` is `undefined` (coming from `_getDependencyEl` -> `fieldEl.dataset.visibilityDependency;`)
The condition was to avoid an error as trying to call `replaceAll()` on `undefined`